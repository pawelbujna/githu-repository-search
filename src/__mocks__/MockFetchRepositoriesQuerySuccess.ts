export const edges = [
    {
        node: {
            id: 'MDEwOlJlcG9zaXRvcnk5MDc1OTkzMA==',
            forkCount: 540,
            stargazerCount: 1138,
            name: 'mock-1',
            url: 'https://github.com/discountry/react',
            description: 'React docs in Chinese | React ‰∏≠ÊñáÊñáÊ°£ÁøªËØë',
            __typename: 'Repository',
        },
        __typename: 'SearchResultItemEdge',
    },
    {
        node: {
            id: 'MDEwOlJlcG9zaXRvcnkxMjE4MTQyMTA=',
            forkCount: 344,
            stargazerCount: 2110,
            name: 'mock-2',
            url: 'https://github.com/primer/react',
            description: "An implementation of GitHub's Primer Design System using React",
            __typename: 'Repository',
        },
        __typename: 'SearchResultItemEdge',
    },
    {
        node: {
            id: 'MDEwOlJlcG9zaXRvcnk3NzUxMzQxOQ==',
            forkCount: 462,
            stargazerCount: 989,
            name: 'mock-3',
            url: 'https://github.com/react-redux-antd-es6/react',
            description: 'Âü∫‰∫éreactÁöÑ‰ºÅ‰∏öÂêéÂè∞ÁÆ°ÁêÜÂºÄÂèëÊ°ÜÊû∂',
            __typename: 'Repository',
        },
        __typename: 'SearchResultItemEdge',
    },
    {
        node: {
            id: 'MDEwOlJlcG9zaXRvcnk3MjYyODI4NQ==',
            forkCount: 437,
            stargazerCount: 792,
            name: 'mock-4',
            url: 'https://github.com/Cathy0807/react',
            description: '‰∫¨‰∏úÈ¶ñÈ°µÊûÑÂª∫',
            __typename: 'Repository',
        },
        __typename: 'SearchResultItemEdge',
    },
    {
        node: {
            id: 'MDEwOlJlcG9zaXRvcnk0MzAyOTI5Ng==',
            forkCount: 182,
            stargazerCount: 221,
            name: 'mock-5',
            url: 'https://github.com/formio/react',
            description: 'JSON powered forms for React.js',
            __typename: 'Repository',
        },
        __typename: 'SearchResultItemEdge',
    },
    {
        node: {
            id: 'MDEwOlJlcG9zaXRvcnkyOTAyODc3NQ==',
            forkCount: 22_637,
            stargazerCount: 106_189,
            name: 'mock-6',
            url: 'https://github.com/facebook/react-native',
            description: 'A framework for building native applications using React',
            __typename: 'Repository',
        },
        __typename: 'SearchResultItemEdge',
    },
    {
        node: {
            id: 'MDEwOlJlcG9zaXRvcnkzNjA2NjI0',
            forkCount: 3564,
            stargazerCount: 19_973,
            name: 'mock-7',
            url: 'https://github.com/ReactiveCocoa/ReactiveCocoa',
            description: 'Cocoa framework and Obj-C dynamism bindings for ReactiveSwift.',
            __typename: 'Repository',
        },
        __typename: 'SearchResultItemEdge',
    },
    {
        node: {
            id: 'MDEwOlJlcG9zaXRvcnkzOTMxNjUzNQ==',
            forkCount: 128,
            stargazerCount: 234,
            name: 'mock-8',
            url: 'https://github.com/azat-co/react',
            description: 'Examples for the React Quickly book. ',
            __typename: 'Repository',
        },
        __typename: 'SearchResultItemEdge',
    },
    {
        node: {
            id: 'MDEwOlJlcG9zaXRvcnk2MTU3MzcwMg==',
            forkCount: 94,
            stargazerCount: 437,
            name: 'mock-9',
            url: 'https://github.com/ui-router/react',
            description: 'üîº UI-Router for React',
            __typename: 'Repository',
        },
        __typename: 'SearchResultItemEdge',
    },
    {
        node: {
            id: 'MDEwOlJlcG9zaXRvcnkxOTg3MjQ1Ng==',
            forkCount: 9623,
            stargazerCount: 48_894,
            name: 'mock-10',
            url: 'https://github.com/remix-run/react-router',
            description: 'Declarative routing for React',
            __typename: 'Repository',
        },
        __typename: 'SearchResultItemEdge',
    },
];

export const MOCK_FETCH_REPOSITORY_QUERY_SUCCESS = {
    data: {
        search: {
            repositoryCount: 3_465_095,
            pageInfo: {
                endCursor: 'Y3Vyc29yOjEw',
                hasNextPage: true,
                hasPreviousPage: false,
                startCursor: 'Y3Vyc29yOjE=',
                __typename: 'PageInfo',
            },
            edges,
            __typename: 'SearchResultItemConnection',
        },
    },
};
