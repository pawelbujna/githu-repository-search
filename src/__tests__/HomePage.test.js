import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { Provider } from 'urql';
import { fromValue } from 'wonka';

import { MOCK_FETCH_REPOSITORY_QUERY_SUCCESS } from '../__mocks__/MockFetchRepositoriesQuerySuccess';
import Home from '../pages/index';

const routerPushMock = jest.fn();

jest.mock('next/router', () => ({
    useRouter() {
        return {
            query: {},
            push: routerPushMock,
        };
    },
}));

const mockClient = {
    executeQuery: () => fromValue(MOCK_FETCH_REPOSITORY_QUERY_SUCCESS),
};

const REPOSITORY_LIST_MOCK = MOCK_FETCH_REPOSITORY_QUERY_SUCCESS.data.search.edges.map(({ node }) => node);

const [firstListElementMock] = REPOSITORY_LIST_MOCK;

const createComponent = () => (
    <Provider value={mockClient}>
        <Home />
    </Provider>
);

const renderHomePage = () => {
    render(createComponent());
};

describe('HomePage', () => {
    it('renders the page', async () => {
        renderHomePage();

        const header = await screen.findByText(firstListElementMock.name);

        expect(header).toBeInTheDocument();
    });

    it('matches the snapshot', async () => {
        const testRenderer = await renderer.create(createComponent());

        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it('should display all elements correctly', async () => {
        renderHomePage();

        const repositoryList = await screen.findAllByTestId('repository-list-item');

        expect(repositoryList.length).toBe(REPOSITORY_LIST_MOCK.length);

        for (const [index, repositoryListItem] of repositoryList.entries()) {
            const { url, stargazerCount, forkCount } = REPOSITORY_LIST_MOCK[index];

            const link = repositoryListItem.querySelector('a');
            const [stargazerElement, forkElement] = repositoryListItem.querySelectorAll('p');

            expect(link).toHaveAttribute('href', url);
            expect(stargazerElement).toHaveTextContent(`🌟 ${stargazerCount}`);
            expect(forkElement).toHaveTextContent(`🍴 ${forkCount}`);
        }
    });

    it('calls router when looking for a different query param', async () => {
        renderHomePage();

        const input = screen.getByPlaceholderText('Type repository name');

        // For some reason 'closest' is not working here
        const submitButton = input.nextSibling.childNodes[0];

        userEvent.type(input, 'freeCodeCamp');
        userEvent.click(submitButton);

        await waitFor(() => {
            expect(routerPushMock).toBeCalled();
        });
    });
});
