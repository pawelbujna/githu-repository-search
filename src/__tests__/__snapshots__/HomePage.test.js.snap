// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches the snapshot 1`] = `
<div>
  <form
    className="ant-form ant-form-horizontal css-dev-only-do-not-override-i56vxb"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="ant-form-item css-dev-only-do-not-override-i56vxb ant-form-item-has-success"
    >
      <div
        className="ant-row ant-form-item-row css-dev-only-do-not-override-i56vxb"
        style={{}}
      >
        <div
          className="ant-col ant-form-item-control css-dev-only-do-not-override-i56vxb"
          style={{}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <span
                className="ant-input-group-wrapper ant-input-search ant-input-search-with-button ant-input-group-wrapper-status-success css-dev-only-do-not-override-i56vxb"
              >
                <span
                  className="ant-input-wrapper ant-input-group css-dev-only-do-not-override-i56vxb"
                >
                  <input
                    className="ant-input ant-input-status-success css-dev-only-do-not-override-i56vxb"
                    hidden={null}
                    id="search"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onCompositionEnd={[Function]}
                    onCompositionStart={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Type repository name"
                    style={null}
                    type="text"
                    value="followers:>100000"
                  />
                  <span
                    className="ant-input-group-addon"
                  >
                    <button
                      className="ant-btn css-dev-only-do-not-override-i56vxb ant-btn-primary ant-input-search-button"
                      disabled={false}
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      type="button"
                    >
                      <span>
                        Submit
                      </span>
                    </button>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <div
    className="ant-list ant-list-sm ant-list-split ant-list-something-after-last-item css-dev-only-do-not-override-i56vxb"
    style={
      {
        "background": "#fff",
        "borderRadius": "4px",
        "padding": "20px",
      }
    }
  >
    <div
      className="ant-spin-nested-loading css-dev-only-do-not-override-i56vxb"
    >
      <div
        className="ant-spin-container"
      >
        <ul
          className="ant-list-items"
        >
          <li
            className="ant-list-item"
            data-testid="repository-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="https://github.com/discountry/react"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    mock-1
                  </a>
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  React docs in Chinese | React ‰∏≠ÊñáÊñáÊ°£ÁøªËØë
                </div>
              </div>
            </div>
            <div
              style={
                {
                  "width": "100px",
                }
              }
            >
              <p>
                üåü 
                1138
              </p>
              <p>
                üç¥ 
                540
              </p>
            </div>
          </li>
          <li
            className="ant-list-item"
            data-testid="repository-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="https://github.com/primer/react"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    mock-2
                  </a>
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  An implementation of GitHub's Primer Design System using React
                </div>
              </div>
            </div>
            <div
              style={
                {
                  "width": "100px",
                }
              }
            >
              <p>
                üåü 
                2110
              </p>
              <p>
                üç¥ 
                344
              </p>
            </div>
          </li>
          <li
            className="ant-list-item"
            data-testid="repository-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="https://github.com/react-redux-antd-es6/react"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    mock-3
                  </a>
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  Âü∫‰∫éreactÁöÑ‰ºÅ‰∏öÂêéÂè∞ÁÆ°ÁêÜÂºÄÂèëÊ°ÜÊû∂
                </div>
              </div>
            </div>
            <div
              style={
                {
                  "width": "100px",
                }
              }
            >
              <p>
                üåü 
                989
              </p>
              <p>
                üç¥ 
                462
              </p>
            </div>
          </li>
          <li
            className="ant-list-item"
            data-testid="repository-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="https://github.com/Cathy0807/react"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    mock-4
                  </a>
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  ‰∫¨‰∏úÈ¶ñÈ°µÊûÑÂª∫
                </div>
              </div>
            </div>
            <div
              style={
                {
                  "width": "100px",
                }
              }
            >
              <p>
                üåü 
                792
              </p>
              <p>
                üç¥ 
                437
              </p>
            </div>
          </li>
          <li
            className="ant-list-item"
            data-testid="repository-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="https://github.com/formio/react"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    mock-5
                  </a>
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  JSON powered forms for React.js
                </div>
              </div>
            </div>
            <div
              style={
                {
                  "width": "100px",
                }
              }
            >
              <p>
                üåü 
                221
              </p>
              <p>
                üç¥ 
                182
              </p>
            </div>
          </li>
          <li
            className="ant-list-item"
            data-testid="repository-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="https://github.com/facebook/react-native"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    mock-6
                  </a>
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  A framework for building native applications using React
                </div>
              </div>
            </div>
            <div
              style={
                {
                  "width": "100px",
                }
              }
            >
              <p>
                üåü 
                106189
              </p>
              <p>
                üç¥ 
                22637
              </p>
            </div>
          </li>
          <li
            className="ant-list-item"
            data-testid="repository-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="https://github.com/ReactiveCocoa/ReactiveCocoa"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    mock-7
                  </a>
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  Cocoa framework and Obj-C dynamism bindings for ReactiveSwift.
                </div>
              </div>
            </div>
            <div
              style={
                {
                  "width": "100px",
                }
              }
            >
              <p>
                üåü 
                19973
              </p>
              <p>
                üç¥ 
                3564
              </p>
            </div>
          </li>
          <li
            className="ant-list-item"
            data-testid="repository-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="https://github.com/azat-co/react"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    mock-8
                  </a>
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  Examples for the React Quickly book. 
                </div>
              </div>
            </div>
            <div
              style={
                {
                  "width": "100px",
                }
              }
            >
              <p>
                üåü 
                234
              </p>
              <p>
                üç¥ 
                128
              </p>
            </div>
          </li>
          <li
            className="ant-list-item"
            data-testid="repository-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="https://github.com/ui-router/react"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    mock-9
                  </a>
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  üîº UI-Router for React
                </div>
              </div>
            </div>
            <div
              style={
                {
                  "width": "100px",
                }
              }
            >
              <p>
                üåü 
                437
              </p>
              <p>
                üç¥ 
                94
              </p>
            </div>
          </li>
          <li
            className="ant-list-item"
            data-testid="repository-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="https://github.com/remix-run/react-router"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    mock-10
                  </a>
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  Declarative routing for React
                </div>
              </div>
            </div>
            <div
              style={
                {
                  "width": "100px",
                }
              }
            >
              <p>
                üåü 
                48894
              </p>
              <p>
                üç¥ 
                9623
              </p>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="ant-list-pagination"
    >
      <ul
        className="ant-pagination css-dev-only-do-not-override-i56vxb"
        style={{}}
      >
        <li
          aria-disabled={true}
          className="ant-pagination-prev ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="left"
              className="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-2"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={2}
        >
          <a
            rel="nofollow"
          >
            2
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-3"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={3}
        >
          <a
            rel="nofollow"
          >
            3
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-4"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={4}
        >
          <a
            rel="nofollow"
          >
            4
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={5}
        >
          <a
            rel="nofollow"
          >
            5
          </a>
        </li>
        <li
          className="ant-pagination-jump-next ant-pagination-jump-next-custom-icon"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="Next 5 Pages"
        >
          <a
            className="ant-pagination-item-link"
          >
            <div
              className="ant-pagination-item-container"
            >
              <span
                aria-label="double-right"
                className="anticon anticon-double-right ant-pagination-item-link-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="double-right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
                  />
                </svg>
              </span>
              <span
                className="ant-pagination-item-ellipsis"
              >
                ‚Ä¢‚Ä¢‚Ä¢
              </span>
            </div>
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-346510"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={346510}
        >
          <a
            rel="nofollow"
          >
            346510
          </a>
        </li>
        <li
          aria-disabled={false}
          className="ant-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={0}
          title="Next Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={false}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="right"
              className="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          className="ant-pagination-options"
        >
          <div
            aria-label="Page Size"
            className="ant-select ant-pagination-options-size-changer css-dev-only-do-not-override-i56vxb ant-select-single ant-select-show-arrow"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
          >
            <div
              className="ant-select-selector"
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-expanded={false}
                  aria-haspopup="listbox"
                  aria-label="Page Size"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autoComplete="off"
                  className="ant-select-selection-search-input"
                  disabled={false}
                  id="rc_select_TEST_OR_SSR"
                  onChange={[Function]}
                  onCompositionEnd={[Function]}
                  onCompositionStart={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onPaste={[Function]}
                  readOnly={true}
                  role="combobox"
                  style={
                    {
                      "opacity": 0,
                    }
                  }
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                className="ant-select-selection-item"
                title="10 / page"
              >
                10 / page
              </span>
            </div>
            <span
              aria-hidden={true}
              className="ant-select-arrow"
              onMouseDown={[Function]}
              style={
                {
                  "WebkitUserSelect": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              <span
                aria-label="down"
                className="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
